CalorieTracker/CalorieTracker/CalorieTrackerViewController.swift

```diff
@@ -0,0 +1,188 @@
+//
+//  ViewController.swift
+//  CalorieTracker
+//
+//  Created by Rob Vance on 8/14/20.
+//  Copyright © 2020 Robs Creations. All rights reserved.
+//
+
+import UIKit
+import SwiftChart
+import CoreData
+
+class CalorieTrackerViewController: UIViewController{
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        updateViews()
+        NotificationCenter.default.addObserver(self, selector: #selector(updateViews), name: .calorieLogChanged, object: nil)
+    }
+    //MARK: - OUTLETS -
+    @IBOutlet weak var tableView: UITableView!
+    @IBOutlet weak var chart: Chart!
+    //MARK: - PROPERTIES -
+    let calorieController = CalorieController()
+    lazy var fetchedResultsController: NSFetchedResultsController<Calorie> = {
+        let fetchRequest: NSFetchRequest<Calorie> = Calorie.fetchRequest()
+        fetchRequest.sortDescriptors = [NSSortDescriptor(key: .timeStamp, ascending: true)]
+        let moc = CoreDataStack.shared.mainContext
+        let frc = NSFetchedResultsController(fetchRequest: fetchRequest,
+                                             managedObjectContext: moc,
+                                             sectionNameKeyPath: nil,
+                                             cacheName: nil)
+        frc.delegate = self
+        do {
+            try frc.performFetch()
+            return frc
+        } catch {
+            fatalError("Error Fetching: \(error)")
+        }
+    }()
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(true)
+        self.viewDidLoad()
+        tableView.reloadData()
+
+    }
+    @objc func updateViews() {
+        let chart = Chart(frame: view.frame)
+        guard let sections = fetchedResultsController.sections else { return }
+        chart.removeAllSeries()
+        var seriesOfDoubles: [Double] = []
+        for section in sections {
+            if let objects = section.objects as? [Calorie] {
+                for object in objects {
+                    seriesOfDoubles.append(Double(object.amount))
+                }
+            }
+        }
+        DispatchQueue.main.async {
+            self.chart.add(ChartSeries(seriesOfDoubles))
+            self.tableView.reloadData()
+        }
+    }
+    //MARK: - IBACTION -
+    @IBAction func addButtonTapped(_ sender: Any) {
+        // Create alert
+        let alert = UIAlertController(title: "Add calorie intake",
+                                      message: "Enter the amount of calories",
+                                      preferredStyle: .alert)
+
+        // Add text field
+        alert.addTextField { textField in
+            textField.placeholder = "Calories"
```

Comment:
In addition to checking for a number, you could set the textfield to use the number pad as input:

``` swift
textField.keyboardType = .numberPad
```